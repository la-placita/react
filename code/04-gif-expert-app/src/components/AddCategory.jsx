import { useState } from 'react';
export const AddCategory = ({ onNewCategory }) => {

    const [ inputValue, setInputValue ] = useState('');
    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue);
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
        <h2>Add Category</h2>
        <input type="text" placeholder="Enter category" value={inputValue} onChange={onInputChange}/>
        <button>Add</button>
        </form>
    );
}