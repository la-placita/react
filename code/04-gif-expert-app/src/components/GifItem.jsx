import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} style={{ width: '200px'}}/>
      <p>{title}</p>
    </div>
  );
}


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
