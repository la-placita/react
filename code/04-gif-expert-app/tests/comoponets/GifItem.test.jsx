import { render, screen} from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
describe('GifItem', () => {
    test('should be match with snapshot', () => {
        const { container } = render(<GifItem title={"Test"} url="https://example.com/test.gif" />);
        expect(container).toMatchSnapshot( );
    });

    test('sholud render the image with the URL and ALT text', () => {
        const title = "Test";
        const url = "https://example.com/test.gif";
        render(<GifItem title={title} url={url} />);
        
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    }); 
});
