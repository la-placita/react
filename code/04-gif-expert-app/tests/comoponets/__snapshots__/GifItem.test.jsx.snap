// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`GifItem should be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test"
      src="https://example.com/test.gif"
      style="width: 200px;"
    />
    <p>
      Test
    </p>
  </div>
</div>
`;
